onnxruntime/core/providers/openvino/ov_versions/capability_2021_1.cc

```diff
@@ -208,6 +208,35 @@ bool IsOpSupported(std::string name, std::string device) {
         supported_ops.insert(supported_ops_cpu.begin(), supported_ops_cpu.end());
      }
     }
+  } else if (device.find("MULTI") == 0) {
+    std::vector<std::string> devices;
+    std::stringstream s_stream(device);
+    while(s_stream.good()) {
+      std::string substr;
+      getline(s_stream, substr, ',');
+      devices.push_back(substr);
+    }
+    if (!common_supported_ops.count(name) == 0) {
+      return true;
+    }
+    for (auto& it : devices) {
+      if(it == "MYRIAD" || "HDDL") {
+        if (supported_ops_vpu.count(name) == 0)  {
+          return false;
+        }
+      }
+      if(it == "GPU") {
+       if (supported_ops_gpu.count(name) == 0)  {
+          return false;
+        }
+      }
+      if(it == "CPU") {
+        if (supported_ops_cpu.count(name) == 0)  {
+          return false;
+        }
+     }
+    }
+    return true;
   }
```

Comment:
this should be same with HETERO mode right? could we merge these with HETERO mode in line 191? if not we should at least make the code consistent, just update the supported_ops for different device type, and let line 241 return the final result.

Reply:
> this should be same with HETERO mode right? could we merge these with HETERO mode in line 191? if not we should at least make the code consistent, just update the supported_ops for different device type, and let line 241 return the final result.

@HectorSVC This isn't the same with HETERO. 
so, in HETERO, we are trying to get the union of all the supported_ops for each of the device that is specified for the Hetero Build. Eventually, at the end, **supported_ops** set will contain the list of all the op's that may be supported by atleast one device from the HETERO device_type we specified. (Hetero works with fallback option)

But for Multi_Device, the op can be only supported if the op is compatible with all the devices specified from the MULTI Device Build type. Basically, we need to be getting the intersection of all the supported_op list's of each device.
The op should be supported on all the devices.

**option 1:**
If we merge both, we would end up having more if checks for MULTI device inside HETERO check.
+
As this check is not a one time check as each and every op in a model has to go through this check, it would be best to have separate check for Hetero and Multi.

**option 2:**
The code can be made consistent but again we will end up adding more if checks for Multi and we need to add additional logic to get the intersection of all the supported_ops device list for all the devices.

with the current logic, it's faster as we check if the op is available in supported list for each device specified in the Multi Device Build type. If  the op is not supported, we directly return false. If it's supported on all the devices, we return true.

Reply:
OK, I see. So, for MULTI mode, the op need to be supported by all the devices specified.
Should we add some UT to cover this method?

---
In reply to: [520298218](https://github.com/microsoft/onnxruntime/pull/5740#discussion_r520298218) [](ancestors = 520298218)

Reply:
@HectorSVC Yes we can add some UT for this method. currently, I see there is no test directory for openvino at **/onnxruntime/test/providers/**  
we need to create one similar to ngraph, tensorrt, etc.. and add the tests there specific to OpenVINO.

There are some other places where we need to add Unit Tests as well for the new features which we merged recently.

I'll discuss with my team and will follow up on that and submit a new PR for adding unit tests.

Reply:
@HectorSVC Just had a discussion with the team. so adding a unit test for MULTI Device feature might be tricky because it is more of a performance edge case. May be we will try to come up with a sample to test out this feature and make it a part of our CI Pipeline as well. Planning to add samples to cover other features like Hetero, batched inference, etc.. as well.